packages/assets-controllers/src/NftController.ts

```diff
@@ -1189,6 +1189,13 @@ export class NftController extends BaseControllerV1<NftConfig, NftState> {
           openSeaEnabled,
           isIpfsGatewayEnabled,
         });
+
+        const needsUpdateNftMetadata =
+          (isIpfsGatewayEnabled && ipfsGateway !== '') || openSeaEnabled;
+
+        if (needsUpdateNftMetadata) {
+          this.updateNftMetadata({ userAddress: selectedAddress });
```

Comment:
Since `updateNftMetadata` is public, removing `nfts` as an argument would be a breaking change. To prevent this, what do you think about moving the lines that you've added to `updateNftMetadata` here?

``` suggestion
          const chainId = this.getCorrectChainId();
          const nfts: Nft[] = this.state.allNfts[userAddress]?.[chainId] ?? [];
          this.updateNftMetadata({ nfts, userAddress: selectedAddress });
```

Reply:
Also can we add tests for this logic? I see some changes made to the tests but I don't see where we are confirming that the NFT metadata is updated when it should be (and, alternatively, where we are confirming that it's not updated when it shouldn't be).

Reply:
Hey @mcmire, thnx for the review! this fct is not used yet by client so i thought its okay to remove the nfts arg. However, have added it back. I think the fct would be more generic if it accepts "nfts" as arg. The behavior could be updated in the future if needed to not only update the NFTs in the state that do not have name/image/description, but to also update the list of nfts passed as input regardless of whether they have name/image/description.